cli/src/main/java/com/box/l10n/mojito/cli/command/PushCommand.java

```diff
@@ -89,7 +100,43 @@ public void execute() throws CommandException {
 
         Repository repository = commandHelper.findRepositoryByName(repositoryParam);
 
-        ArrayList<FileMatch> sourceFileMatches = commandHelper.getSourceFileMatches(commandDirectories, fileType, sourceLocale, sourcePathFilterRegex);
+        if (commitHash != null && StringUtils.isBlank(commitHash)) {
+            throw new IllegalArgumentException(
+                    MessageFormat.format(
+                            "The value provided with the {0} parameter can not be blank/whitespace.",
+                            Param.COMMIT_HASH_LONG));
+        }
+
+        if (commitHash != null && !recordPushRun) {
+            throw new IllegalArgumentException(
+                    MessageFormat.format(
+                            "Whenever {0} is provided, {1} must also be passed in as true.",
+                            Param.COMMIT_HASH_LONG,
+                            Param.RECORD_PUSH_RUN_LONG));
+        }
+
+        if (recordPushRun && commitHash == null) {
+            throw new IllegalArgumentException(
+                    MessageFormat.format(
+                            "Whenever {0} is set to true, {1} must also be passed in with a commit hash.",
+                            Param.RECORD_PUSH_RUN_LONG,
+                            Param.COMMIT_HASH_LONG));
+        }
+
+        if (recordPushRun && !StringUtils.isWhitespace(commitHash) && pushType == PushService.PushType.SEND_ASSET_NO_WAIT_NO_DELETE) {
```

Comment:
Do we need the !StringUtils.isWhitespace(commitHash) check here as it's contained in the first check already?

Reply:
Indeed, removed it for the next iteration.